openedx/core/djangoapps/programs/tasks.py

```diff
@@ -4,12 +4,112 @@
 
 from celery import task
 from celery.utils.log import get_task_logger  # pylint: disable=no-name-in-module, import-error
+from django.contrib.auth.models import User
+from edx_rest_api_client.client import EdxRestApiClient
+
+from lms.djangoapps.certificates.api import get_certificates_for_user, is_passing_status
+
+from openedx.core.djangoapps.credentials.models import CredentialsApiConfig
+from openedx.core.djangoapps.credentials.utils import get_user_credentials
+from openedx.core.djangoapps.programs.models import ProgramsApiConfig
+from openedx.core.lib.token_utils import get_id_token
 
-from lms.djangoapps.certificates.api import get_certificates_for_user
 
 LOGGER = get_task_logger(__name__)
 
 
+def get_api_client(api_config, student):
+    """
+    Create and configure an API client for authenticated HTTP requests.
+
+    Args:
+        api_config: ProgramsApiConfig or CredentialsApiConfig object
+        student: User object as whom to authenticate to the API
+
+    Returns:
+        EdxRestApiClient
+
+    """
+    id_token = get_id_token(student, api_config.OAUTH2_CLIENT_NAME)
+    return EdxRestApiClient(api_config.internal_api_url, jwt=id_token)
+
+
+def get_completed_courses(student):
+    """
+    Determine which courses have been completed by the user.
+
+    Args:
+        student:
+            User object representing the student
+
+    Returns:
+        iterable of dicts with structure {'course_id': course_key, 'mode': cert_type}
+
+    """
+    all_certs = get_certificates_for_user(student.username)
+    return [
+        {'course_id': cert['course_key'], 'mode': cert['type']}
+        for cert in all_certs
+        if is_passing_status(cert['status'])
+    ]
+
+
+def get_completed_programs(client, course_certificates):
+    """
+    Given a set of completed courses, determine which programs are completed.
+
+    Args:
+        client:
+            programs API client (EdxRestApiClient)
+        course_certificates:
+            iterable of dicts with structure {'course_id': course_key, 'mode': cert_type}
+
+    Returns:
+        list of program ids
+
+    """
+    return client.programs.complete.post(course_certificates)
```

Comment:
The program completion endpoint returns [an object](https://github.com/edx/programs/blob/master/programs/apps/api/v1/views.py#L130), not a list. This will end up returning a dictionary as-is.